eos-tech-support/eos-write-usb-internet

```diff
@@ -0,0 +1,243 @@
+    />
+</library>
+XML
+)
+
+args=$(getopt -o t:h \
+    -l tmp-dir: \
+    -l debug \
+    -l help \
+    -n "$0" -- "$@")
+eval set -- "$args"
+
+tmp_dir=/tmp/eos-usb-internet-writer-tmp
+debug=
+
+usage() {
+    local self
+    self=$(basename "$0")
+    cat <<EOF
+Usage:
+    $self [options] -- IMAGE_OPTIONS
+
+Arguments:
+    IMAGE_OPTIONS       Everything after -- is forwarded to the $eos_write_live_image
+                        command, appended to the $self command internal options (which
+                        are the options required to copy the Software and content files
+                        to the USB key).
+
+Options:
+    -t, --tmp-dir PATH  Temporary directory to use for downloading files and
+                        save the USB key files to copy, user is in charge of
+                        cleaning this directory if required, the command will
+                        try to reuse all the content from this directory.
+                        (default: $tmp_dir)
+        --debug         Enable debugging output
+    -h, --help          Show this message
+EOF
+}
+
+while true; do
+    case "$1" in
+        -t|--tmp-dir)
+            shift
+            tmp_dir="${1%/}"
+            shift
+            ;;
+        --debug)
+            shift
+            set -x
+            debug=--debug
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+done
+
+workspace_dir="$tmp_dir/workspace"
+downloads_dir="$tmp_dir/downloads"
+
+if ! hash apfs-fuse 2>/dev/null; then
```

Comment:
Can we ship this in Endless OS?

Reply:
It looks like this is not packaged for Debian or Ubuntu already. Debian ships **libfsapfs-utils**, which includes a different fuse apfs implementation, `fpafsmount`. Would that work for what we are trying to achieve here? If so, that would add ~2MB to the ostree, so it should be ok to ship it, in which case we should have a separate commit in this same PR including it.